pkgs/misc/emulators/yuzu/base.nix

```diff
@@ -0,0 +1,66 @@
+{ pname, version, src, branchName
+, stdenv, fetchFromGitHub, wrapQtAppsHook
+, cmake, pkgconfig
+, libpulseaudio, libjack2, alsaLib, sndio, ecasound
+, vulkan-loader, vulkan-headers
+, qtbase, qtwebengine, qttools
+, nlohmann_json, rapidjson
+, zlib, zstd, libzip, lz4
+, glslang
+, boost173
+, catch2
+, fmt
+, SDL2
+, udev
+, libusb1
+, ffmpeg
+}:
+
+stdenv.mkDerivation rec {
+  inherit pname version src;
+
+  nativeBuildInputs = [ cmake pkgconfig wrapQtAppsHook ];
+  buildInputs = [
+    libpulseaudio libjack2 alsaLib sndio ecasound
+    vulkan-loader vulkan-headers
+    qtbase qtwebengine qttools
+    nlohmann_json rapidjson
+    zlib zstd libzip lz4
+    glslang
+    boost173
+    catch2
+    fmt
+    SDL2
+    udev
+    libusb1
+    ffmpeg
+  ];
+
+  cmakeFlags = [ "-DENABLE_QT_TRANSLATION=ON" "-DYUZU_USE_QT_WEB_ENGINE=ON" "-DUSE_DISCORD_PRESENCE=ON" ];
+
+  # Trick the configure system. This prevents a check for submodule directories.
+  preConfigure = "rm -f .gitmodules";
```

Comment:
```suggestion
  preConfigure = ''
    rm .gitmodules
  '';
```

Reply:
The `-f` flag here is required, as some branches (early-access here) do not have `.gitmodules` present. Without the `-f` flag, this would cause `rm` to fail.